hydra/_internal/utils.py

```diff
@@ -239,7 +239,9 @@ def run_hydra(
                 config_name=config_name, overrides=args.overrides, cfg_type=args.cfg
             )
         elif args.shell_completion:
-            hydra.shell_completion(config_name=config_name, overrides=args.overrides)
+            hydra.shell_completion(
```

Comment:
Can you explain the changes you are making here?

Reply:
A new command line option `--_completion` is added. For `bash`, it would be `None`. For fish, its value would be the current command line until where the user presses TAB, for example `python ./foo.py db` if the user presses TAB after `db`. This modification passed `args._completion` into specific shell plugins (`hydra.shell_completion() -> plugin.query(...)`).

Reply:
Does fish support registering completion functions like bash completion?
If it does, we can have the adapting logic in that shell function (it can populate $COMP_LINE) and then we wouldn't need the different interface for completion.